### PR TITLE
[prometheus]: Fix statefulSet PVC volume mount name

### PR DESCRIPTION
**Pull request by naphta**
#### What this PR does / why we need it
ping: @gianrubio @zanhsieh @Xtigyro @naseemkullah

I inadvertently left a bug in my [last PR.](https://github.com/prometheus-community/helm-charts/pull/3289)

#### Which issue this PR fixes
N/A

#### Special notes for your reviewer
Looks like I forgot to change the volume mount name, this PR updates the volume mount to rename the mounted volume in-line with the name override. 

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus/templates/sts.yaml [hunk 1]
```diff
@@ -203,7 +203,7 @@ spec:
           volumeMounts:
             - name: config-volume
               mountPath: /etc/config
-            - name: storage-volume
+            - name: {{ default "storage-volume" .Values.server.persistentVolume.statefulSetNameOverride "" }}
```
Comment (zeritti) [inline charts/prometheus/templates/sts.yaml:206]: Looks good, thank you. I'd suggest, though, to remove the third argument to the `default` function as the function is happy with just two args.

Comment (naphta) [reply charts/prometheus/templates/sts.yaml:206]: @zeritti I've changed it over to a pipe on the original code, and wrapped it in a more defensive if statement if for whatever reason someone passes in `.server.persistentVolume.enabled=false` and  sets `.server.persistentVolume.statefulSetNameOverride`

Comment (zeritti) [reply charts/prometheus/templates/sts.yaml]: Thank you, looks good.

File: charts/prometheus/templates/sts.yaml [hunk 2]
```diff
@@ -203,7 +203,11 @@ spec:
           volumeMounts:
             - name: config-volume
               mountPath: /etc/config
+            {{- if and .Values.server.persistentVolume.enabled .Values.server.persistentVolume.statefulSetNameOverride }}
+            - name: {{ .Values.server.persistentVolume.statefulSetNameOverride }}
+            {{- else }}
             - name: storage-volume
+            {{- end }}
```
Comment (zanhsieh) [inline charts/prometheus/templates/sts.yaml:210]:
```suggestion
            - name: {{ ternary .Values.server.persistentVolume.statefulSetNameOverride "storage-volume" (and .Values.server.persistentVolume.enabled .Values.server.persistentVolume.statefulSetNameOverride) }}
```

File: charts/prometheus/Chart.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 name: prometheus
 appVersion: v2.43.0
-version: 21.1.2
+version: 21.1.3
```
Comment (zanhsieh) [inline charts/prometheus/Chart.yaml:4]:
```suggestion
version: 22.0.1
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**